src/content/reference/react/useOptimistic.md

```diff
@@ -0,0 +1,202 @@
+---
+title: useOptimistic
+canary: true
+---
+
+<Canary>
+
+The `useOptimistic` Hook is currently only available in React's canary and experimental channels. Learn more about [React's release channels here](/community/versioning-policy#all-release-channels).
+
+</Canary>
+
+<Intro>
+
+`useOptimistic` is a React Hook that lets you optimistically update the UI.
+
+```js
+  const [optimisticState, addOptimistic] = useOptimistic(state, updateFn);
+```
+
+</Intro>
+
+<InlineToc />
+
+---
+
+## Reference {/*reference*/}
+
+### `useOptimistic(state, updateFn)` {/*use*/}
+
+`useOptimistic` is a React Hook that lets you optimistically update the UI.
+
+```js
+import { useOptimistic } from 'react';
+
+function AppContainer() {
+  const [optimisticState, addOptimistic] = useOptimistic(
+    state,
+    // updateFn
+    (currentState, newState) => {
+      // merge and return new state
+    }
+  );
+}
+```
+
+[See more examples below.](#usage)
+
+#### Parameters {/*parameters*/}
+
+* `state`: `state` is the source of truth so if the `state` is ever changed and we get a new value in from the server, the return value `optimisticState` will be set the that too since it will always treat `state` as the final source of truth.
+* `updateFn`: `updateFn` is the mutation that will occur to `optimisticState` when `addOptimistic` is called. `updateFn` takes in two parameters. The `currentState` and the `newState`. The return value will be the merged value of the `currentState` and `newState`.
+
+
+#### Returns {/*returns*/}
+
+* `optimisticState`: `optimisticState` is the optimistic state, it will default to what `state` is.
+* `addOptimistic`: `addOptimistic` is the dispatching function to call that will run what we define in `updateFn`.
+
+---
+
+## Usage {/*usage*/}
+
+### Optimistically update the UI while waiting for a network request {/*optimistically-update-while-waiting-for-network*/}
+
+<Sandpack>
+
+```js like-button.js active
+import { useOptimistic } from "react";
+
+export function LikeButton({ likes, updateLikes, disabled }) {
+  const [optimisticLikes, addOptimisticLike] = useOptimistic(
+    likes,
+    (likes) => likes + 1
+  );
+
+  async function addLike() {
+    addOptimisticLike();
+    await updateLikes(likes);
+  }
+
+  return (
+    <button disabled={disabled} onClick={addLike}>
```

Comment:
It looks like we need to wrap this in a transition, that might actually be a good caveat
<img width="998" alt="CleanShot 2023-10-25 at 18 14 58@2x" src="https://github.com/reactjs/react.dev/assets/1309636/50160c20-71b3-407a-8016-8222adc59780">

Reply:
I'll remove this example for now, and update it with the caveat and transition